api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/IndividualListController.java

```diff
@@ -216,9 +219,12 @@ private static IndividualListResults getResultsForVClassQuery(IndividualListQuer
 
     private static IndividualListQueryResults buildAndExecuteVClassQuery(
 			List<String> vclassURIs, String alpha, Locale locale, int page,
-			int pageSize, IndividualDao indDao)
+			int pageSize, VitroRequest vreq)
 			throws SearchEngineException {
+		 IndividualDao indDao = vreq.getWebappDaoFactory().getIndividualDao();
 		 SearchQuery query = SearchQueryUtils.getQuery(vclassURIs, alpha, locale, page, pageSize);
+		 Set<String> currentRoles = SearchFiltering.getCurrentUserRoles(vreq);
```

Comment:
I think this method (buildAndExecuteVClassQuery) is not used from some loop, if you think this might be used in iterations for the same user we should consider inviting `SearchFiltering.getCurrentUserRoles(vreq); ` out of this method, just one time and used as a parameter for this method.

Reply:
I think it is not used in loops, so I think performance should be ok.